### PR TITLE
Accept batched tensor of images as input to image processor

### LINKED ISSUE #21142: Error when passing a tensor of images to CLIPProcessor
### System Info

- huggingface_hub version: 0.11.1
- Platform: Linux-4.19.0-23-cloud-amd64-x86_64-with-glibc2.31
- Python version: 3.10.8
- Running in iPython ?: No
- Running in notebook ?: No
- Running in Google Colab ?: No
- Token path ?: /root/.huggingface/token
- Has saved token ?: False
- Configured git credential helpers: !f()
- FastAI: N/A
- Tensorflow: 2.11.0
- Torch: 1.13.1
- Jinja2: 3.1.2
- Graphviz: N/A
- Pydot: N/A

### Who can help?

@ArthurZucker @amyeroberts @NielsRogge 

### Information

- [ ] The official example scripts
- [X] My own modified scripts

### Tasks

- [ ] An officially supported task in the `examples` folder (such as GLUE/SQuAD, ...)
- [X] My own task or dataset (give details below)

### Reproduction

1. Run the following script:

```python
from transformers import CLIPProcessor
import torch
model_name_or_path = "openai/clip-vit-large-patch14"
processor: CLIPProcessor = CLIPProcessor.from_pretrained(
            model_name_or_path
        )

dummy_input = torch.randn(10, 3, 224, 224)

dummy_output = processor(images=dummy_input, return_tensors="pt")
```

2. Look at the monitor to see the error: 
```
 ---------------------------------------------------------------------------
KeyError                                  Traceback (most recent call last)
File /opt/conda/envs/main/lib/python3.10/site-packages/PIL/Image.py:2953, in fromarray(obj, mode)
   2952 try:
-> 2953     mode, rawmode = _fromarray_typemap[typekey]
   2954 except KeyError as e:

KeyError: ((1, 1, 224, 224), '|u1')

The above exception was the direct cause of the following exception:

TypeError                                 Traceback (most recent call last)
Cell In[2], line 10
      4 processor: CLIPProcessor = CLIPProcessor.from_pretrained(
      5             model_name_or_path
      6         )
      8 dummy_input = torch.randn(10, 3, 224, 224)
---> 10 dummy_output = processor(images=dummy_input, return_tensors="pt")

File /opt/conda/envs/main/lib/python3.10/site-packages/transformers/models/clip/processing_clip.py:85, in CLIPProcessor.__call__(self, text, images, return_tensors, **kwargs)
     82     encoding = self.tokenizer(text, return_tensors=return_tensors, **kwargs)
     84 if images is not None:
---> 85     image_features = self.feature_extractor(images, return_tensors=return_tensors, **kwargs)
     87 if text is not None and images is not None:
     88     encoding["pixel_values"] = image_features.pixel_values
...
-> 2955         raise TypeError("Cannot handle this data type: %s, %s" % typekey) from e
   2956 else:
   2957     rawmode = mode

TypeError: Cannot handle this data type: (1, 1, 224, 224), |u1
```

### Expected behavior

The function should return a preprocessed tensor containing a batch of images.

### PR DESCRIPTION
**Pull request by amyeroberts**
# What does this PR do?

Adds functionality to `image_utils` so that a batched tensor of images can be accepted as input to the image processors. 

Fixes #21142 #14650

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [x] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/main/CONTRIBUTING.md#start-contributing-pull-requests),
      Pull Request section?
- [x] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
- [ ] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/main/docs#writing-source-documentation).
- [x] Did you write any new necessary tests?

### CODE REVIEW

File: src/transformers/image_utils.py [hunk 1]
```diff
@@ -91,6 +91,45 @@ def is_batched(img):
...
+
+    # Either the input is a single image, in which case we create a list of length 1
+    if isinstance(images, PIL.Image.Image):
+        # PIL images are never batched
+        return [images]
+
+    if is_valid_image(images):
+        if images.ndim == expected_ndims + 1:
+            # Batch of images
+            images = [image for image in images]
```
Comment (AntreasAntoniou) [inline src/transformers/image_utils.py:117]: You could use images.unbind(0) instead as it's more efficient.

Comment (amyeroberts) [reply src/transformers/image_utils.py:117]: @AntreasAntoniou - yep, thanks for the suggestion! We just need to add a check on the type, as this function accepts numpy, tensorflow, jax and pytorch tensors.

Comment (amyeroberts) [reply src/transformers/image_utils.py:117]: Decided to leave as it for the moment as including means adding a bunch of if/else logic that isn't super pretty for a small speed up ~1µs

```
In [1]: import numpy as np

In [2]: import torch

In [3]: pt_imgs = torch.randint(0, 256, (16, 3, 20, 20))

In [4]: %timeit [img for img in pt_imgs] 
9.44 µs ± 104 ns per loop (mean ± std. dev. of 7 runs, 100,000 loops each)

In [5]: %timeit pt_imgs.unbind(0)
8.67 µs ± 113 ns per loop (mean ± std. dev. of 7 runs, 100,000 loops each)
```

**Review by sgugger — APPROVED** [review]
Thanks for working on this!

**Pull request merged**